src/core/mac/mac_filter.cpp

```diff
@@ -44,46 +44,46 @@ namespace Mac {
 
 Filter::Filter(void)
     : mAddressMode(OT_MAC_FILTER_ADDRESS_MODE_DISABLED)
-    , mRssIn(OT_MAC_FILTER_FIXED_RSS_DISABLED)
+    , mDefaultRssIn(kFixedRssDisabled)
 {
-    for (int i = 0; i < GetMaxEntries(); i++)
-    {
-        memset(&mEntries[i], 0, sizeof(Entry));
+    FilterEntry *entry = &mFilterEntries[0];
 
-        mEntries[i].mFiltered = false;
-        mEntries[i].mRssIn    = OT_MAC_FILTER_FIXED_RSS_DISABLED;
+    for (size_t num = OT_ARRAY_LENGTH(mFilterEntries); num != 0; num--, entry++)
+    {
+        entry->mFiltered = false;
+        entry->mRssIn    = kFixedRssDisabled;
     }
 }
 
-Filter::Entry *Filter::FindEntry(const ExtAddress &aExtAddress)
+Filter::FilterEntry *Filter::FindEntry(const ExtAddress &aExtAddress)
 {
-    Entry *entry = NULL;
+    FilterEntry *entry = &mFilterEntries[0];
 
-    for (uint8_t i = 0; i < GetMaxEntries(); i++)
+    for (size_t num = OT_ARRAY_LENGTH(mFilterEntries); num != 0; num--, entry++)
```

Comment:
How about something like:
```c++
FilterEntry *entry = &mFilterEntries[0];
FilterEntry *end   = entry + OT_ARRAY_LENGTH(mFilterEntries);

for (; entry < end; entry++)
{
    ...
}
```

Reply:
I tend to over-optimize sometimes... :)... (compare with zero vs pointer check). 
Should not really matter. Changes them in new pushed commit.